### ISSUE TITLE
Feature request: Add SparseCategoricalFocalCrossentropy

### ISSUE THREAD

**Issue by ishitta-iyer** [keras-team/keras#23442]
## Summary

Add sparse-label support for categorical focal crossentropy through:

- `keras.losses.sparse_categorical_focal_crossentropy`
- `keras.losses.SparseCategoricalFocalCrossentropy`

Keras currently provides `CategoricalFocalCrossentropy`, which expects one-hot labels, and `SparseCategoricalCrossentropy`, which accepts integer labels. There is no corresponding focal loss for sparse categorical labels.

## Motivation

Sparse labels are common in semantic segmentation, token classification, and multiclass problems with many classes. Converting these labels to one-hot tensors solely to use focal loss increases memory usage and adds unnecessary preprocessing.

A native sparse implementation would provide focal loss while preserving integer labels and Keras' backend-agnostic behavior.

## Proposed API

```python
keras.losses.SparseCategoricalFocalCrossentropy(
    alpha=0.25,
    gamma=2.0,
    from_logits=False,
    ignore_class=None,
    axis=-1,
    reduction="sum_over_batch_size",
    name="sparse_categorical_focal_crossentropy",
    dtype=None,
)
```

I would like to work on this if the contribution makes sense.

**Comment by dghnts** [comment]
Hi, I'm interested in contributing to this if it becomes available. Happy to help if the current assignee isn't able to continue, or with any related follow-up work.

**Comment by jeffcarp** [comment]
Thanks for the proposal! I think this makes sense. @ishitta-iyer since you proposed this you're welcome to send a PR if still interested.

**Comment by ishitta-iyer** [comment]
@jeffcarp Yes perfect! I'll have a PR ready in a bit!

**Comment by ishitta-iyer** [comment]
added pr #23482 with an implementation. let me know if there are any changes to be made @jeffcarp

**Comment by MahadIrfan01** [comment]
Hello, is this issue still available to work on?
